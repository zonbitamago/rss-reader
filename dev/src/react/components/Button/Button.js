import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./Button.css";
import MaterialButton from "@material-ui/core/Button";

const StyleYes = {
  background: "#03A9F4",
  color: "white",
  borderRadius: "5px"
};

const StyleNo = {
  background: "#FF1744",
  color: "white",
  borderRadius: "5px"
};

class Button extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    if (this.props.type == "yes") {
      return (
        <div className="Button">
          <MaterialButton
            id="yes"
            onClick={this.props.handleClick}
            variant="raised"
            style={StyleYes}
          >
            Yes
          </MaterialButton>
        </div>
      );
    } else if (this.props.type == "no") {
      return (
        <div className="Button">
          <MaterialButton
            id="no"
            onClick={this.props.handleClick}
            variant="raised"
            style={StyleNo}
          >
            No
          </MaterialButton>
        </div>
      );
    }
  }
}

Button.propTypes = {
  type: PropTypes.string.isRequired,
  handleClick: PropTypes.func
};

Button.defaultProps = {};

export default Button;
