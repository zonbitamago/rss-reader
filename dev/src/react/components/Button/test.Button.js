import React from "react";
import { shallow } from "enzyme";

import Button from "./Button";

describe("Button", () => {
  let component, props, component2, props2;

  beforeEach(() => {
    props = { type: "yes" };
    component = shallow(<Button {...props} />);
    props2 = { type: "no" };
    component2 = shallow(<Button {...props} />);
  });

  it("should", () => {
    expect(component).toMatchSnapshot();
    expect(component2).toMatchSnapshot();
  });
});
