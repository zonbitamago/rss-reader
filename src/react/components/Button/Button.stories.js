import React from "react";
import { storiesOf } from "@storybook/react";

import Button from "./Button";

storiesOf("Atoms/Button", module)
  .add("Yes", () => <Button type="yes" />)
  .add("No", () => <Button type="no" />);
