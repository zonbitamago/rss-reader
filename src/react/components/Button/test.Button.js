import React from "react";
import { shallow } from "enzyme";

import Button from "./Button";

describe("Button yes", () => {
  let component, props;

  beforeEach(() => {
    props = { type: "yes" };
    component = shallow(<Button {...props} />);
  });

  it("should", () => {
    expect(component).toMatchSnapshot();
  });
});

describe("Button no", () => {
  let component, props;

  beforeEach(() => {
    props = { type: "no" };
    component = shallow(<Button {...props} />);
  });

  it("should", () => {
    expect(component).toMatchSnapshot();
  });
});

describe("Button else", () => {
  let component, props;

  beforeEach(() => {
    props = { type: "else" };
    component = shallow(<Button {...props} />);
  });

  it("should", () => {
    expect(component).toMatchSnapshot();
  });
});
