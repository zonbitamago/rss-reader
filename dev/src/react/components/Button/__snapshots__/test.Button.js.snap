// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    type="yes"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Button)
        style={
          Object {
            "background": "#448AFF",
            "borderRadius": "5px",
            "color": "white",
          }
        }
        variant="raised"
      >
        Yes
      </WithStyles(Button)>,
      "className": "Button",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Yes",
        "style": Object {
          "background": "#448AFF",
          "borderRadius": "5px",
          "color": "white",
        },
        "variant": "raised",
      },
      "ref": null,
      "rendered": "Yes",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Button)
          style={
            Object {
              "background": "#448AFF",
              "borderRadius": "5px",
              "color": "white",
            }
          }
          variant="raised"
        >
          Yes
        </WithStyles(Button)>,
        "className": "Button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Yes",
          "style": Object {
            "background": "#448AFF",
            "borderRadius": "5px",
            "color": "white",
          },
          "variant": "raised",
        },
        "ref": null,
        "rendered": "Yes",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Button should 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    type="yes"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Button)
        style={
          Object {
            "background": "#448AFF",
            "borderRadius": "5px",
            "color": "white",
          }
        }
        variant="raised"
      >
        Yes
      </WithStyles(Button)>,
      "className": "Button",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Yes",
        "style": Object {
          "background": "#448AFF",
          "borderRadius": "5px",
          "color": "white",
        },
        "variant": "raised",
      },
      "ref": null,
      "rendered": "Yes",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Button)
          style={
            Object {
              "background": "#448AFF",
              "borderRadius": "5px",
              "color": "white",
            }
          }
          variant="raised"
        >
          Yes
        </WithStyles(Button)>,
        "className": "Button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Yes",
          "style": Object {
            "background": "#448AFF",
            "borderRadius": "5px",
            "color": "white",
          },
          "variant": "raised",
        },
        "ref": null,
        "rendered": "Yes",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
